Make libdrm_amdgpu optional.

https://gitlab.com/coolercontrol/coolercontrol/-/commit/c6d87d24af3710ac7a3f60f84bd2e2a3c7faca93

From 9cef588b2d3fc50dbd40be451f7a78bd34f45b6a Mon Sep 17 00:00:00 2001
From: Alfred Wingate <parona@protonmail.com>
Date: Mon, 23 Sep 2024 14:35:35 +0300
Subject: [PATCH] Allow disabling linking to libdrm_amdgpu

Notably users on Gentoo may choose to disable building libdrm with amdgpu support leading to a build
failure.

Signed-off-by: Alfred Wingate <parona@protonmail.com>
--- a/coolercontrold/Cargo.toml
+++ b/coolercontrold/Cargo.toml
@@ -123,6 +123,9 @@ test-context = "0.3.0"
 tempfile = "3.14.0"
 #rstest = "0.15.0"  # possibility for the future
 
+[features]
+libdrm_amdgpu = []
+
 [profile.release]
 strip = "symbols"
 opt-level = 3
--- a/coolercontrold/src/repositories/gpu/amd.rs
+++ b/coolercontrold/src/repositories/gpu/amd.rs
@@ -24,6 +24,7 @@ use std::sync::Arc;
 
 use anyhow::{anyhow, Context, Result};
 use heck::ToTitleCase;
+#[cfg(feature = "libdrm_amdgpu")]
 use libdrm_amdgpu_sys::AMDGPU::{DeviceHandle, GPU_INFO};
 use log::{error, info, trace, warn};
 use regex::Regex;
@@ -93,7 +94,11 @@ impl GpuAMD {
                 Err(err) => error!("Error initializing AMD Hwmon Freqs: {}", err),
             };
             let fan_curve_info = Self::get_fan_curve_info(&device_path).await;
+            #[cfg(feature = "libdrm_amdgpu")]
             let drm_device_name = Self::get_drm_device_name(&path);
+            #[cfg(not(feature = "libdrm_amdgpu"))]
+            // https://github.com/Umio-Yasuno/libdrm-amdgpu-sys-rs/blob/c10593dd00ad822d71e284424455023a3ad6a20c/amdgpu/mod.rs#L14
+            let drm_device_name = Some(String::from("AMD Radeon Graphics"));
             let pci_device_names = devices::get_device_pci_names(&path);
             let model = devices::get_device_model_name(&path)
                 .or(drm_device_name)
@@ -139,6 +144,7 @@ impl GpuAMD {
         }
     }
 
+    #[cfg(feature = "libdrm_amdgpu")]
     fn get_drm_device_name(base_path: &Path) -> Option<String> {
         let slot_name = devices::get_pci_slot_name(base_path)?;
         let path = format!("/dev/dri/by-path/pci-{slot_name}-render");
-- 
2.47.1

